### DISCUSSION TITLE
Starting with Frigate 0.12.0 you can use the official image on DSM 7

### DISCUSSION THREAD

**Discussion by deviant77** [weltenwort/frigate-synology-dsm7#95] [General]
libusb is now [built without udev support](https://github.com/blakeblackshear/frigate/pull/3768/files) in the official 0.12.0 Frigate image. :partying_face:

I'm currently running the latest dev image on DSM 7, but 0.12.0 is close to a beta release.

**Comment by weltenwort** [comment]
That's great, thanks for sharing!

**Reply by Caros2017** [reply]
Very nice, thanks for sharing indeed. I just ran some tests on `0.12.0 Beta 1` and it seems to work almost perfectly :)

They now even included `openvino`, which means that even without the `Google Coral` you can use the GPU from the Synology NAS. (If ofcourse not using the new Ryzen) Have you been testing this and if so, what's the experience?

The only thing I don't seem to get working in the beta is the logging of `intel-VAAPI` under `system->GPUs`. I get the following message:
```
intel-vaapi
There was an error getting usage stats. Either your GPU does not support this or frigate does not have proper access.
```

I also don't get this working when I run the container as `privileged: true`

**Comment by deviant77** [comment]
[0.12.0 Beta 2](https://github.com/blakeblackshear/frigate/discussions/4948) was just released.

**Comment by cchewanik** [comment]
@deviant77 @Caros2017 are you able to screenshot your Synology docker image settings / variables / etc., I am getting errors when starting up my image

Thank you!

**Comment by Floriszz** [comment]
I use the DSM Task Scheduler to run Frigate on DSM 6.x flawless with following docker commands now, but want to update DSM to 7.x eventually. I rather not use any CLI/SSH or Docker Compose to install/run/update Frigate on my Synology NAS DS981+ with Google Coral USB. 
Can you confirm that I can do the same with DSM 7.x in combination with Frigate 0.12.0+ with  the Google Coral USB (Edge TPU) ?

The bold parts are different from: https://docs.frigate.video/frigate/installation/#docker 

docker run -d \
  --name frigate \
  --restart=unless-stopped \
  --mount type=tmpfs,target=/tmp/cache,tmpfs-size=1000000000 \
  --device /dev/dri/renderD128 \
  --shm-size=64m \
  --privileged \
  -v /path/to/your/storage:/media/frigate \
  -v /path/to/your/config.yml:/config/config.yml:ro \
  -v /etc/localtime:/etc/localtime:ro \
  -**v /dev/bus/usb:/dev/bus/usb** \
  -v /path/to/your/db:/db \
  -e FRIGATE_RTSP_PASSWORD='password' \
  -p **5055**:5000 \
  -p 1935:1935 \
  blakeblackshear/frigate:stable

**Reply by mx-vatte** [reply]
I can confirm this way with the official image frigate 0.12.0 (see [https://github.com/weltenwort/frigate-synology-dsm7/discussions/95#discussioncomment-5728379](url))

**Comment by kimbo1st** [comment]
Official 0.12.0 version is out... can we update or we need to wait till we see a new release here?

**Reply by Caros2017** [reply]
The drivers are inside the official container now. No need anymore for this one.

**Reply by kimbo1st** [reply]
Ok so this compiled version isn't required anymore.
So I just download the image, stop the container, clear and start back up.

**Reply by mx-vatte** [reply]
Did you do so or was your last sentence a question? For me it is! Is this the procedure to follow?

**Reply by kimbo1st** [reply]
Well I have tried and it stayed the same version.
So I have downloaded the new image and made a new docker. .. 
modified the Config file to add go2rtc which replaces rtmp. Doesn’t startup giving errors on the Config.. :(

**Comment by danuals** [comment]
I’ve updated with the official release. Works perfectly. Config needs adjusting so read release notes. Kind RegardsDanOn 16 Apr 2023, at 8:09 pm, kimbo1st ***@***.***> wrote:﻿
Official 0.12.0 version is out... can we update or we need to wait till we see a new release here?

**Comment by imne** [comment]
Got 0.12.0 working, this is my config
Port Settings & environmental variables
![image](https://user-images.githubusercontent.com/4516296/232566027-93529af4-539d-4678-861f-d247a42661d3.png)

Volume
![image](https://user-images.githubusercontent.com/4516296/232566080-d9488ab5-84b8-4b02-bb40-64d0e71429a0.png)

![image](https://user-images.githubusercontent.com/4516296/232566237-aafcb6cb-530e-42bd-9a82-b109225e67cb.png)

it's running fine, I'm using the exact same config from the 0.11.1
[config.zip](https://github.com/weltenwort/frigate-synology-dsm7/files/11253956/config.zip)

with the above config everything works fine on my end on 0.12.0 can see all 10 of my cameras and it's recording events fine 
   
![image](https://user-images.githubusercontent.com/4516296/232568824-9591e08f-97b2-45d1-9140-5eae57977577.png)

**Reply by imne** [reply]
also my home assistant frigate integration still works 100%
![image](https://user-images.githubusercontent.com/4516296/232570382-f53dd47b-4790-42d1-8673-6f6787704a2e.png)

**Reply by kimbo1st** [reply]
Nice and got it working too now, just the image out of the box works fine.
Also set the cameras to go2rtc and this works smoother, it seems to use less cpu power. Have also the Google coral.

**Reply by imne** [reply]
updated the official Frigate documentation with the set by step guide to make it easier for people 
https://docs.frigate.video/frigate/installation#synology-nas-on-dsm-7

**Reply by luigitare** [reply]
My biggest challenge was downloading the image into the Docker GUI.

For others like me: SSH into the Syno:

    Run "sudo docker pull ghcr.io/blakeblackshear/frigate:stable"

    Select the image "/blakeblackshear/frigate:stable" and click "Start"

    Configure settings, see: https://docs.frigate.video/frigate/installation/#synology-nas-on-dsm-7

**Reply by mx-vatte** [reply]
Thx for this hint.

I managed to use frigate 0.12.0 by using the Synology task scheduler on my DS923+ (Ryzen1600).

docker run -d \
--name frigate_v12 \
--restart=unless-stopped \
--mount type=tmpfs,target=/tmp/cache,tmpfs-size=1000000000 \
--device /dev/bus/usb:/dev/bus/usb \
--shm-size=640m \
--privileged \
-v /volume1/docker/frigate/storage:/media/frigate \
-v /volume1/docker/frigate/config.yml:/config/config.yml \
-v /etc/localtime:/etc/localtime:ro \
-v /dev/bus/usb:/dev/bus/usb \
-e FRIGATE_RTSP_PASSWORD='yourpassword' \
-p 8001:5000 \
-p 8554:8554 \
-p 8555:8555/tcp \
-p 8555:8555/udp \
ghcr.io/blakeblackshear/frigate:stable

Kind regards,
Vatte

**Reply by Floriszz** [reply]
Very important detail in this taskabove is the link to the image. While Synology Docker can not pull from Github Container Registry through the UI, you can using the task scheduler. Just make sure it starts with: 'ghcr.io/' as shown in the commands above.
Thanks a million for this!

**Reply by Floriszz** [reply]
Another important addition is that every line, but not the last has to end with \ . (at least for DSM6.x. I wanted to test Frigate 12 first on my DSM6.x, before upgrade to DSM7 so I knew for sure script and config were working. 
So the above would look like:
docker run -d \ 
--name frigate_v12 \ 
--restart=unless-stopped \ 
--mount type=tmpfs,target=/tmp/cache,tmpfs-size=1000000000 \ 
--device /dev/bus/usb:/dev/bus/usb \ 
--shm-size=640m \ 
--privileged \ 
-v /volume1/docker/frigate/storage:/media/frigate \ 
-v /volume1/docker/frigate/config.yml:/config/config.yml \ 
-v /etc/localtime:/etc/localtime:ro \ 
-v /dev/bus/usb:/dev/bus/usb \ 
-e FRIGATE_RTSP_PASSWORD='yourpassword' \ 
-p 8001:5000 \ 
-p 8554:8554 \ 
-p 8555:8555/tcp \ 
-p 8555:8555/udp \ 
ghcr.io/blakeblackshear/frigate:stable

**Comment by baldisos** [comment]
Can confirm this is working with the official image on DSM 7. Huge thanks to @weltenwort and all contributors tho, you guys saved me a lot of hassle with this project for as long as it was necessary. ♥️

**Reply by weltenwort** [reply]
Thanks for the kind words :heart:

**Comment by MichaelH02** [comment]
I have the same issue as Caros2017 described on Jan 7.
I'm using a Synology DS920+ with DSM 7.2 and frigate in Version 0.12.1

In the log I found the following error:
frigate.util ERROR : Unable to poll GPU stats: Failed to initialize PMU! (Invalid argument)


Has anybody an idea how to fix that problem?



I installed frigate in docker with a compose file.

version: "3.9"
services:
  frigate:
    container_name: frigate
    privileged: true # this may not be necessary for all setups
    restart: unless-stopped
    image: ghcr.io/blakeblackshear/frigate:stable
    shm_size: "512mb" # update for your cameras based on calculation above
    devices:
      - /dev/bus/usb:/dev/bus/usb # passes the USB Coral, needs to be modified for other versions
      - /dev/dri:/dev/dri # for intel hwaccel, needs to be updated for your hardware
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - /volume1/docker/frigate/config/config.yml:/config/config.yml
      - /volume1/docker/frigate/media:/media/frigate
      - type: tmpfs # Optional: 1GB of memory, reduces SSD/SD Card wear
        target: /tmp/cache
        tmpfs:
          size: 1000000000
    ports:
      - "5000:5000"
      - "8554:8554" # RTSP feeds
      - "8555:8555/tcp" # WebRTC over tcp
      - "8555:8555/udp" # WebRTC over udp
    environment:
      FRIGATE_RTSP_PASSWORD: "password"
      LIBVA_DRIVER_NAME: "i965"



My frigate config is the following:

mqtt:
  enabled: False

database:
  path: /config/frigate.db

cameras:
  Kamera_1:
    ffmpeg:
      hwaccel_args: preset-vaapi
      inputs:
        - path: rtsp://user:password@172.19.16.110:554/stream1
          roles:
            - detect
            - record
    detect:
      enabled: True
      width: 1920
      height: 1080
      fps: 15
    objects:
      track:
      - cat
    snapshots:
      enabled: True
      timestamp: False
      bounding_box: True
      retain:
        default: 3
    record:
      enabled: true
      retain:
        days: 1
        mode: motion
      events:
        retain:
          default: 1
          mode: active_objects
        pre_capture: 5
        post_capture: 8
        objects:
          - cat

  Kamera_2:
    ffmpeg:
      hwaccel_args: preset-vaapi
      inputs:
        - path: rtsp://user:password@172.19.16.114:554/stream1
          roles:
            - detect
            - record
    detect:
      enabled: True
      width: 2304
      height: 1296
      fps: 15
    objects:
      track:
      - person
      - cat
    snapshots:
      enabled: True
      timestamp: False
      bounding_box: True
      retain:
        default: 3
    record:
      enabled: true
      retain:
        days: 1
        mode: motion
      events:
        retain:
          default: 1
          mode: active_objects
        pre_capture: 5
        post_capture: 8
        objects:
          - person
          - cat

detectors:
  coral:
    type: edgetpu
    device: usb

**Reply by danuals** [reply]
Synology blocks it, can only ignore it and not get stats unfortunately.

**Reply by MichaelH02** [reply]
Thank you. Let's see if there is a solution in the future
